### PR TITLE
JavaScript capstone project - Your API-based webapp

### PR DESCRIPTION
**Pull request by ntzwilly**
## Description

- Use JavaScript to make websites dynamic
- Use callbacks and promises to consume Two APIs: Meals DB & Involvement API
- Use Webpack
- Write units tests for a JavaScript app.
- Follow Gitflow.
- Send and receive data from an API.
- Use API documentation.
- Perform a code review for a team member.

### CODE REVIEW

File: src/comments.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import getCommentsCount from './utils';
+
+export const displayComments = (items) => {
```
Comment (ZawadiSumba66) [inline src/comments.js:3]: - [ ] When a user first adds a comment to the popup there is an undefined word appears and the comments count does not increase. It's only after one closes and reopens the popup again that the undefined disappears and the comments count increases. 
- [ ] There is also another issue where if create another comment in the same popup the previous one disappears.
- [ ] Check out this[ loom](https://www.loom.com/share/7f35ba213980400b988346721535828e) video to further demonstrate your app behavior.

File: src/utils.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const getCommentsCount = (comments) => comments.length || 0;
```
Comment (ZawadiSumba66) [inline src/utils.js:1]: - [ ] As from the requirements given, you need to create a specific function to calculate counter numbers on each page i.e for the comments and the items; even if the API provides that information. You can write the function in such a way that you loop through the data fetched and then display the count.

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+const header = elementGenerator('header');
+const logo = elementGenerator('img', 'logo');
+logo.src = logoImage;
+const navigation = elementGenerator('nav');
+const uList = elementGenerator('ul');
+const listOne = elementGenerator('li', 'spaceship');
+const linkOne = elementGenerator('a');
+linkOne.href = '#';
+
+async function mealCounter() {
```
Comment (ZawadiSumba66) [inline src/index.js:90]: - [ ] Ii seems that you have forgotten to apply the ES6 syntax on some function declarations i.e the use of arrow functions here. Kindly correct this and all the other files/areas where you had missed this.

**Review by ZawadiSumba66 — CHANGES_REQUESTED** [review]
Hi Team 😃 , 

Good job so far on this project 👏 👏 
However, there are some issues that you still need to work on to go to the next step but you are almost there! 💪 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/comments.js (hunk 1 above)
Comment (ntzwilly) [reply src/comments.js:3]: The issues are now fixed

File: src/utils.js (hunk 2 above)
Comment (ntzwilly) [reply src/utils.js:1]: Thank you, I have created a function

File: src/index.js (hunk 3 above)
Comment (ntzwilly) [reply src/index.js:90]: I am now using ES6

File: README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+## Built With
+
+- javaScript
+- CSS
+- Webpack
+
+## Live Demo
+
+[Live Demo Link](https://ntzwilly.github.io/JS-capstone/dist/)
+
```
Comment (dasileker) [inline README.md:18]: - [ ] Th live demo is different from the one in the dev branch, please update it.

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+      const heart = elementGenerator('img');
+      heart.src = icon;
+      heart.alt = 'heart-image';
+
+      const like = elementGenerator('p');
+      like.classList.add('small');
+      like.textContent = ' 0 likes';
+
+      heart.addEventListener('click', async (e) => {
+        e.preventDefault();
```
Comment (dasileker) [inline src/index.js:155]: - [ ] The Likes are not working, I do see that the previous TSE mentioned this but still not working see screenshot 👇

https://user-images.githubusercontent.com/61562626/129460027-1aa2a068-f216-48d0-969a-10b8e865205e.mp4

**Review by dasileker — CHANGES_REQUESTED** [review]
**### Hi @ntzwilly,** 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hemant-soni-vst-au4 — APPROVED** [review]
Hi @ntzwilly and @Adewunmi97, 

Good work on making the requested changes. 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Alt Text

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**